doozer/doozerlib/k_distgit.py

```diff
@@ -0,0 +1,84 @@
+import os
+import pathlib
+from typing import Union
+from dockerfile_parse import DockerfileParser
+from artcommonlib import assertion, logutil, build_util, exectools
+from artcommonlib.pushd import Dir
+from doozerlib.distgit import ImageDistGitRepo
+from doozerlib.metadata import Metadata
+
+
+class KonfluxImageDistGitRepo(ImageDistGitRepo):
+    """
+    It's not technically distgit anymore, but using the same name for simplicity
+    """
+    def __init__(self, metadata, autoclone=True):
+        super(KonfluxImageDistGitRepo, self).__init__(metadata, autoclone=False)
+
+        # Using k_distgits_dir which points to the new konflux dir
+        self.distgit_dir = os.path.join(self.runtime.k_distgits_dir, self.metadata.distgit_key)
+        self.dg_path = pathlib.Path(self.distgit_dir)
+
+        if autoclone:
+            self.clone()
+
+    def clone(self) -> None:
+        # Refresh if dir already exists, instead of cloning
+        if os.path.isdir(self.distgit_dir):
+            with Dir(self.distgit_dir):
+                self.logger.info(f"{self.distgit_dir} dir already exists, refreshing git instead of cloning")
+                exectools.cmd_assert('git fetch --all', retries=3)
+                exectools.cmd_assert('git reset --hard @{upstream}', retries=3)
+            return
+
+        branch = self.metadata.config.content.source.git.branch.target
+        url = self.metadata.config.content.source.git.url
+
+        git_args = ["--no-single-branch", "--branch", branch]
+        self.runtime.git_clone(url, self.distgit_dir, gitargs=git_args)
+
+    def push(self) -> Union[Metadata, bool]:
+        """
+        Push to the appropriate branch on openshift-priv
+        """
+        # Figure out which branch to push to
+        branch = f"art-<{self.runtime.group}>-assembly-<{self.runtime.assembly}>-dgk-<{self.name}>"
```

Comment:
How about `art:{self.runtime.group}:assembly:{self.runtime.assembly}:dgk:{self.name}`

Reply:
`fatal: 'art:openshift-4.17:assembly:test:dgk:openshift-base-rhel9' is not a valid branch name`